google-api-client/src/main/java/com/google/api/client/googleapis/util/Utils.java

```diff
@@ -55,6 +71,82 @@ public static HttpTransport getDefaultTransport() {
     static final HttpTransport INSTANCE = new NetHttpTransport();
   }
 
+  /**
+   * Returns the `cert_provider_command` field in context_aware_metadata.json file.
+   * 
+   * @param contextAwareMetadataJson ~/.secureConnect/context_aware_metadata.json file content.
+   * @return `cert_provider_command` field
+   * @since 1.31
+   */
+  @SuppressWarnings("unchecked")
+  public static ArrayList<String> extractCertificateProviderCommand(String contextAwareMetadataJson) {
+    GsonBuilder builder = new GsonBuilder();
+    LinkedTreeMap<String, Object> map = (LinkedTreeMap<String, Object>)builder.create().fromJson(contextAwareMetadataJson, Object.class);
+    return (ArrayList<String>)map.get("cert_provider_command");
+  }
+
+  /**
+   * Returns the default client certificate by running the cert_provider_command commands
+   * from ~/.secureConnect/context_aware_metadata.json file.
+   * 
+   * @return The default client certificate input stream
+   * @since 1.31
+   */
+  public static InputStream loadDefaultCertificate() throws IOException, GeneralSecurityException {
```

Comment:
Maybe `loadDefaultMtlsCertificate`? This is a general `Utils` class.

Reply:
done